utils/wasm-gen/src/generator/syscalls/invocator.rs

```diff
@@ -315,6 +329,49 @@ impl<'a, 'b> SyscallsInvocator<'a, 'b> {
         };
         instructions.append(&mut result_processing);
 
+        if let Some(wait_frequency) = self
+            .config
+            .waiting_frequency()
+            .filter(|_| invocable.is_wait_syscall())
+        {
+            let mem_size = self.memory_size_in_bytes();
+
+            let upper_limit = mem_size.saturating_sub(1) as i32;
+            let wait_called_ptr = upper_limit.saturating_sub(50);
```

Comment:
This actually means that every syscall invocation must never touch that memory. Can we assert that?

Reply:
805fa95b0bb72a0efba37a115abcb38d853964d3